pkg/apis/apps/validation/validation.go

```diff
@@ -75,6 +75,40 @@ func ValidateStatefulSetSpec(spec *apps.StatefulSetSpec, fldPath *field.Path) fi
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("podManagementPolicy"), spec.PodManagementPolicy, fmt.Sprintf("must be '%s' or '%s'", apps.OrderedReadyPodManagement, apps.ParallelPodManagement)))
 	}
 
+	switch spec.UpdateStrategy.Type {
+	case "":
+		allErrs = append(allErrs, field.Required(fldPath.Child("updateStrategy"), ""))
+	case apps.OnDeleteStatefulSetStrategyType, apps.RollingUpdateStatefulSetStrategyType:
```

Comment:
I turned them off because it might be confusing. I'm open to suggestion about the best practice wrt to leaving the su- field allowable. I gave it some thought previously and thought that ignoring it might be the best course of action to allow easy type changes. After further thought it seems like it might be confusing. A user could get a SatefulSet and see a partition for rolling update and wonder why it isn't respected. During the design phase we kicked the idea around of applying a partition directly in the spec and we landed on using the structs. I'm leaning in favor of failing validation if the a partition is applied to a strategy that can not support it. It does not represent a valid API object based on the design.